### PR TITLE
Ed: fix bug on vj extension creation

### PR DESCRIPTION
**Pull request by antoine-de**
we don't want to do an extension for vj split over the dst

we cannot do a good check on the validity pattern because of overnight
trips

DO NOT MERGE for the moment, I need to fix another issue on our sherbrooke data set to be able to test it

### CODE REVIEW

**Comment by antoine-de** [comment]
should fix #819 

thanks @skywave for the tips !

**Comment by antoine-de** [comment]
seems to be working on our sherbrooke dataset (with #859 )

File: source/routing/raptor.cpp [hunk 1]
```diff
@@ -62,6 +62,10 @@ bool RAPTOR::apply_vj_extension(const Visitor& v, const bool global_pruning,
     bool add_vj = false;
     bool result = false;
     while(vj) {
+        if (! vj->validity_pattern->check(DateTimeUtils::date(workingDt))) {
```
Comment (TeXitoi) [inline source/routing/raptor.cpp]: It should be tested with is_valid_day 6 lines behind. And you don't check adatped if needed.

File: source/ed/data.cpp [hunk 2]
```diff
@@ -70,14 +70,17 @@ void Data::build_block_id() {
...
-            }
+              [](const types::VehicleJourney* vj1, const types::VehicleJourney* vj2) {
+        if(vj1->block_id != vj2->block_id) {
+            return vj1->block_id < vj2->block_id;
+        } else if (vj1->utc_to_local_offset != vj2->utc_to_local_offset) {
+            // we don't want to link the splited vjs
+            return vj1->utc_to_local_offset < vj2->utc_to_local_offset;
+        } else {
+            return vj1->stop_time_list.back()->arrival_time <=
+                    vj2->stop_time_list.front()->departure_time;
```
Comment (TeXitoi) [inline source/ed/data.cpp:81]: I prefere

``` c++
 return vj1->stop_time_list.front()->departure_time <= vj2->stop_time_list.front()->departure_time;
```

as the current version is not a strict weak ordering. The succession is already tested line 100.

**Pull request merged**